File: TeemoAttacking.go
/**
 * LeetCode 495 Teemo Attacking
 */
func findPoisonedDuration(timeSeries []int, duration int) int {
	if timeSeries == nil || len(timeSeries) == 0 || duration < 0 {
		panic("param error")
	}
	total := 0
	for i := 0;i < len(timeSeries) - 1;i++ {
		total += min(duration,timeSeries[i + 1] - timeSeries[i])
	}
	return total + duration
}

func min(a int,b int) int {
	if a < b {
		return a
	}
	return b
}

File: TeemoAttacking.java
/**
 * LeetCode 495 Teemo Attacking
 */
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        if (timeSeries == null || timeSeries.length == 0) {
            return 0;
        }
        int res = 0;
        for (int i = 0; i < timeSeries.length - 1; i++) {
            if (timeSeries[i] + duration >= timeSeries[i + 1]) {
                res += timeSeries[i + 1] - timeSeries[i];
            } else {
                res += duration;
            }
        }
        res += duration;
        return res;
    }
}
